cmd/sdkstats/main.go

```diff
@@ -7,46 +7,141 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"strings"
 
 	"github.com/go-github-metrics-1/pkg/sdkstats"
-
 	"github.com/google/go-github/github"
 	"golang.org/x/oauth2"
 )
 
+type helmVersionParser struct{}
+type baseVersionParser struct {
+	searchQ      string
+	searchLatest string
+}
+type gomodVersionParser struct {
+	searchQ string
+}
+type unknownVersionParser struct{}
+
 func main() {
 	var token string
 	flag.StringVar(&token, "token", "", "GitHub API token")
 	flag.Parse()
-	args := flag.Args()
+	if len(token) == 0 {
+		log.Fatal("GITHUB API TOKEN MUST BE ENTERED \n Usage: './main --token=YOURTOKEN'")
+	}
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: args[0]},
+		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := github.NewClient(tc)
+	//client := github.NewClient(tc)
 
-	var queries []string
-
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/helm-operator")
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/ansible-operator")
-	queries = append(queries, "filename:go.mod github.com/operator-framework/operator-sdk")
-	queries = append(queries, "filename:Gopkg.toml github.com/operator-framework/operator-sdk")
+	queries := []sdkstats.RepoMetadataQuery{
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "ansible",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/ansible-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ:      "quay.io/operator-framework/ansible-operator:v",
+				searchLatest: "quay.io/operator-framework/ansible-operator:master",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "helm",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/helm-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ:      "quay.io/operator-framework/helm-operator:v",
+				searchLatest: "quay.io/operator-framework/helm-operator:latest",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "go.mod",
+			Queries: []string{
+				"filename:go.mod github.com/operator-framework/operator-sdk",
+			},
+			VersionParser: &gomodVersionParser{
+				searchQ: "replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v",
```

Comment:
We should not assume the SDK version will be in it's own replace statement. It could look like any of the following:
```
# Option 1:
replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk <version>

# Option 2
replace (
    ...
    github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk <version>
    ...
)

# Option 3
require (
    ...
    github.com/operator-framework/operator-sdk <version>
    ...
)
```

With these three options in mind, the pattern is:
`<repo><whitespace><version><newline>`

We can probably craft a regex for this as well.